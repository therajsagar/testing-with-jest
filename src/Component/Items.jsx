import React from 'react';
import PropTypes from 'prop-types';

/**
 * Render a list of items
 * @param {Object} props - List of items
 */
export default function Items(props) {
  const { items = [] } = props;

  if (!items.length) {
    // No Items on the list, render an empty message
    return <span className='empty-message'>No items in list</span>;
  }

  if (items.length === 1) {
    // One Item in the list, render a span
    return <span className='item-message'>{items[0]}</span>;
  }

  // Multiple items on the list, render a list
  return (
    <ul>
      {items.map(item => (
        <li key={item} className='item-message'>
          {item}
        </li>
      ))}
    </ul>
  );
}

Items.propTypes = {
  items: PropTypes.array.isRequired
};

Items.defaultProps = {
  items: []
};
