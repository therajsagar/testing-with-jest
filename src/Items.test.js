import React from 'react';
import { create } from 'react-test-renderer';
import Items from './Component/Items';

describe(`ItemList Test Suite`, () => {
  it('renders correctly when there are no items', () => {
    const tree = create(<Items />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when there is one item', () => {
    const items = ['one'];
    const tree = create(<Items items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when there are multiple items', () => {
    const items = ['one', 'two', 'three'];
    const tree = create(<Items items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
