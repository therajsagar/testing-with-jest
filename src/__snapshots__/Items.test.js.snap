// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList Test Suite renders correctly when there are multiple items 1`] = `
<ul>
  <li
    className="item-message"
  >
    one
  </li>
  <li
    className="item-message"
  >
    two
  </li>
  <li
    className="item-message"
  >
    three
  </li>
</ul>
`;

exports[`ItemList Test Suite renders correctly when there are no items 1`] = `
<span
  className="empty-message"
>
  No items in list
</span>
`;

exports[`ItemList Test Suite renders correctly when there is one item 1`] = `
<span
  className="item-message"
>
  one
</span>
`;
